src/app/components/text-input-field.tsx

```diff
@@ -67,15 +72,15 @@ export function TextInputField({
         flexDirection="column"
         htmlFor={name}
         justifyContent="center"
```

Comment:
I'm not proud of this `variant === 'mnemonic'` stuff but it lets us re-use most of the `TextInputField` code without risking breaking other areas

Reply:
Why not just duplicate it? IMO duplication in some our our components is a much lesser crime than breaking the separation of concerns between a generic UI component and specific domain use-case.

Reply:
OK, I can do that. I was trying to avoid duplicating all of the logic also but you're right, it's worth it here